scipy/special/tests/test_boxcox.py

```diff
@@ -104,3 +105,21 @@ def test_inv_boxcox1p_underflow():
     y = inv_boxcox1p(x, lam)
     assert_allclose(y, x, rtol=1e-14)
 
+
+@pytest.mark.parametrize(
+    "x, lmb",
+    [[100, 155],
+     [0.01, -155]]
+)
+def test_boxcox_premature_overflow(x, lmb):
+    # test boxcox & inv_boxcox
+    y = boxcox(x, lmb)
+    assert np.isfinite(y)
+    x_inv = inv_boxcox(y, lmb)
+    assert_almost_equal(x, x_inv)
+
+    # test boxcox1p & inv_boxcox1p
+    y1p = boxcox1p(x-1, lmb)
+    assert np.isfinite(y1p)
+    x1p_inv = inv_boxcox1p(y1p, lmb)
+    assert_almost_equal(x-1, x1p_inv)
```

Comment:
Drive by comment, all new tests should use `assert_allclose`
```suggestion
    assert_allclose(x, x_inv)

    # test boxcox1p & inv_boxcox1p
    y1p = boxcox1p(x-1, lmb)
    assert np.isfinite(y1p)
    x1p_inv = inv_boxcox1p(y1p, lmb)
    assert_allclose(x-1, x1p_inv)
```